src/transaction/types/input.rs

```diff
@@ -174,8 +194,32 @@ impl Input {
             _ => None,
         }
     }
+
+    #[cfg(feature = "std")]
+    pub fn validate_signature(txhash: &Bytes32, owner: &Address, witness: &Witness) -> bool {
```

Comment:
Why not use `&self.owner` instead?

Reply:
We can't always have `self.owner` because `Input` also has the variant `Contract`. If we take `self`, the user could call `validate_signature` also for a contract, and that wouldn't validate anything. The intent of taking the `owner` instead of `self` is to translate this restriction clearly in the API

Reply:
API changed in https://github.com/FuelLabs/fuel-tx/commit/a77f4e84eb40e75a806e03c41ce44285b6a18bdc